### PR TITLE
Us4 node parsing

### PR DESCRIPTION
**Pull request by JonathanBush**
Abstract classes for BinaryOperator and UnaryOperator.

### CODE REVIEW

File: src/backend/BinaryOperator.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    super();
+    this.leftOperand = null;
+    this.rightOperand = null;
+    this.operatorText = '';
+    if (this.constructor === BinaryOperator) {
+      throw new TypeError('Abstract class \'BinaryOperator\' cannot be instantiated');
+    }
+  }
+
+  getText() {
```
Comment (zlmonroe) [inline src/backend/BinaryOperator.js:22]: getText should be a member of the BinaryOperator class

File: src/backend/Node.js [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
+ *
+ * @date 9/18/2019
+ */
+class Node {
+  constructor() {
+    if (this.constructor === Node) {
+      throw new TypeError('Abstract class \'Node\' cannot be instantiated');
+    }
+
+    if (this.getText === undefined) {
```
Comment (zlmonroe) [inline src/backend/Node.js:13]: This will still be undefined b/c getText is not in the class

File: src/backend/UnaryOperator.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  constructor() {
+    super();
+    this.operand = null;
+    this.operatorText = '';
+    if (this.constructor === UnaryOperator) {
+      throw new TypeError('Abstract class \'UnaryOperator\' cannot be instantiated');
+    }
+  }
+
+  getText() {
```
Comment (zlmonroe) [inline src/backend/UnaryOperator.js:21]: Same as above comments about getText

File: src/backend/BinaryOperator.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/**
```
Comment (zlmonroe) [inline src/backend/BinaryOperator.js:1]: We might want to create a folder/module for nodes under backend instead of throwing everything directly under.

**Review by zlmonroe — CHANGES_REQUESTED** [review]

File: src/backend/BinaryOperator.js (hunk 4 above)
Comment (mattfrautnick) [reply src/backend/BinaryOperator.js:1]: Agree!

**Review by JonathanBush — COMMENTED** [review]
Cleared up confusion from indentation level.

**Pull request merged**